### PR TITLE
Fix help prompts in custom Server Config

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/15399
Also fixes awkward sizing of the modal.

Before (same dialog for both help buttons):
image

After:
image
image

Copy on both could do with improvements.

### CODE REVIEW

File: src/components/views/auth/ServerConfig.js [hunk 1]
```diff
@@ -244,7 +266,7 @@ export default class ServerConfig extends React.PureComponent {
         });
         return <div className={classes}>
             {_t("Enter your custom identity server URL <a>What does this mean?</a>", {}, {
-                a: sub => <a className="mx_ServerConfig_help" href="#" onClick={this.showHelpPopup}>
+                a: sub => <a className="mx_ServerConfig_help" href="#" onClick={this.showIsHelpPopup}>
```
Comment (turt2live) [inline src/components/views/auth/ServerConfig.js:269]: please let's stop using `is` as a shorthand for `identity server` - `identity` is just as fine, and doesn't come across as booleans.

File: src/components/views/auth/ServerConfig.js [hunk 2]
```diff
@@ -211,16 +212,37 @@ export default class ServerConfig extends React.PureComponent {
...
+            button: _t("Dismiss"),
+        });
+    };
+
+    showIsHelpPopup = () => {
+        Modal.createTrackedDialog('What does this mean?', 'IS', InfoDialog, {
+            title: _t("Identity Server URL"),
+            description: _t(
+                "The optional identity server is responsible for allowing you " +
+                "to find people by their e-mail address.",
```
Comment (turt2live) [inline src/components/views/auth/ServerConfig.js:234]: and phone number

**Review by turt2live — CHANGES_REQUESTED** [review]

File: src/components/views/auth/ServerConfig.js (hunk 2 above)
Comment (t3chguy) [reply src/components/views/auth/ServerConfig.js:234]: https://github.com/vector-im/element-web/issues/3711

Comment (turt2live) [reply src/components/views/auth/ServerConfig.js:234]: well, it also handles your pending 3pid invites - it's not just discovery.

File: src/components/views/auth/ServerConfig.js [hunk 3]
```diff
@@ -226,12 +226,12 @@ export default class ServerConfig extends React.PureComponent {
...
     };
 
-    showIsHelpPopup = () => {
+    showIdentityServerHelpPopup = () => {
         Modal.createTrackedDialog('What does this mean?', 'IS', InfoDialog, {
             title: _t("Identity Server URL"),
             description: _t(
                 "The optional identity server is responsible for allowing you " +
-                "to find people by their e-mail address.",
+                "to find people by their e-mail address and phone number.",
```
Comment (turt2live) [inline src/components/views/auth/ServerConfig.js:234]: still no mention of pending 3pid invites for your own account :(

**Review by turt2live — APPROVED** [review]
lgtm code-wise - the copy iteration is probably best done by people who know words and not me

**Review by nadonomy — CHANGES_REQUESTED** [review]
Blocking this on design— adding to the list to figure out how to expose this in a manner which doesn't add more cognitive load (e.g. using a (?) icon).

**Comment by t3chguy** [comment]
Superseded by https://github.com/matrix-org/matrix-react-sdk/pull/5426

**Pull request closed without merging**